common/workunit/workunit.cpp

```diff
@@ -5462,9 +5490,70 @@ class CDaliWorkUnitFactory : public CWorkUnitFactory, implements IDaliClientShut
                 }
                 if (!iter)
                     return NULL;
-                sortElements(iter, sortOrder.get(), nameFilterLo.get(), nameFilterHi.get(), unknownAttributes, elements);
+
+                Owned<IPropertyTreeIterator> lWWUPTreeIter = createWorkunitPTreeIterator(iter);
+                sortElements(lWWUPTreeIter ? lWWUPTreeIter : iter, sortOrder.get(), nameFilterLo.get(), nameFilterHi.get(), unknownAttributes, elements);
                 return conn.getClear();
             }
+            IPropertyTreeIterator* createWorkunitPTreeIterator(IPropertyTreeIterator* originalIter)
+            {
+                Owned<IRemoteConnection> conWorkUnitAborts = querySDS().connect("/WorkUnitAborts/", myProcessSession(), 0, SDS_LOCK_TIMEOUT);
+                if (!conWorkUnitAborts)
+                    return nullptr;
+
+                bool hasDynamicAbortingState = false;
+                Owned<IPropertyTree> lWWUPTreeRoot = createPTree("lWWUs");
+                IPropertyTree* wuAbortsPTree = conWorkUnitAborts->queryRoot();
+                ForEach(*originalIter)
+                {
+                    IPropertyTree& cur = originalIter->query();
+
+                    bool dynamicAbortingState = false;
+                    const char* wuid = cur.queryName();
+                    WUState state = getState(&cur, wuAbortsPTree, dynamicAbortingState);
```

Comment:
perhaps slightly cleaner if you removed the boolean here and in definition and in getWUState etc., and simply returned state, then code as:
```
    WUState state = (WUState) getEnum(&cur, "@state", states);
    WUState dynamicAbortingState = getState(&cur, wuAbortsPTree);
    if (state == dynamicAbortingState)
        lWWUPTreeRoot->addPropTree(wuid, LINK(&cur));
...
```

Reply:
minor: this getState() function which mainly defines a lambda function, could be inlined here as:
```
    auto isAbortingFunc = [wuid, wuAbortsPTree] () { return wuAbortsPTree->hasProp(wuid); };
```
so above code becomes:
```
                    auto isAbortingFunc = [wuid, wuAbortsPTree] () { return wuAbortsPTree->hasProp(wuid); };

                    WUState state = (WUState) getEnum(&cur, "@state", states);
                    WUState dynamicAbortingState = getWUState(&cur, isAbortingFunc);
                    if (state == dynamicAbortingState)
                        lWWUPTreeRoot->addPropTree(wuid, LINK(&cur));
```

Reply:
Will change